packages/server/src/__tests__/ApolloServer.test.ts

```diff
@@ -141,6 +141,40 @@ describe('ApolloServer construction', () => {
     // @ts-expect-error
     takesConfig({ modules: [] });
   });
+  describe('with configuration options', () => {
+    it('test', async () => {
+      const server = new ApolloServer({
+        typeDefs,
+        resolvers,
+        stringifyResult: (value: FormattedExecutionResult) => {
+          let result = JSON.stringify(value, null, 10000);
+          result = result.replace('world', 'stringifyResults works!'); // replace text with something custom
+          return result;
+        },
+      });
+
+      await server.start();
+
+      const request = {
+        httpGraphQLRequest: {
+          method: 'POST',
+          headers: new HeaderMap([['content-type', 'application-json']]),
+          body: { query: '{ hello }' },
+          search: '',
+        },
+        context: async () => ({}),
+      };
+
+      const { body } = await server.executeHTTPGraphQLRequest(request);
+      expect(JSON.stringify(body)).toMatchInlineSnapshot(
+        `"{\"kind\":\"complete\",\"string\":\"{\\n          \\\"data\\\": {\\n                    \\\"hello\\\": \\\"stringifyResults works!\\\"\\n          }\\n}\"}"`,
+      );
+      // const result = singleResult(body);
+      // expect(result.errors).toBeUndefined();
+      // expect(result.data?.hello).toBe('world');
```

Comment:
It's a little weird to stringify the response body for snapshotting, I think this is what you're looking for and we get a better visual of the result (included some cleanup of stray code).

```suggestion
      const { body } = await server.executeHTTPGraphQLRequest(request);
      assert(body.kind === 'complete');
      expect(body.string).toMatchInlineSnapshot(`
        "{
                  "data": {
                            "hello": "stringifyResults works!"
                  }
        }"
      `);
```

Reply:
Gotcha, this is what i was hoping would happen. But, I didn't realize i needed this line:

```ts
      assert(body.kind === 'complete');
```